### ISSUE TITLE
headless chrome can't capture screenvideo or animation

### ISSUE THREAD

**Issue by t-io** [ChromeDevTools/devtools-protocol#17]
I try to capture some animations from a website and stitch them together using ffmpeg.
As far as I understand the docs [startScreencast](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-startScreencast) is the way to go.

If I understand that right I can start the screencast with 
`await Page.startScreencast({format: 'png', everyNthFrame: 1});` 
and listen to __every__ incoming frame with 
```
Page.screencastFrame( image =>{
    const {data, metadata} = image;
    console.log(metadata);
});
```
But it's never printed out something. So I assume it's not triggered.

I archived my goal with something like this:
```
let counter = 0;
while(counter < 500){
    await Page.startScreencast({format: 'png', everyNthFrame: 1});
    const {data, metadata} = await Page.screencastFrame();
    console.log(metadata);
          counter += 1;
        }
````
Which feels like a non-performant hack.
So any suggestions on how to use `startScreencast` and `screencastFrame` properly?

**Comment by auchenberg** [comment]
Try to call `Page.enable` before subscribing to the events. https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-enable

**Comment by t-io** [comment]
@auchenberg thanks for the quick response. 
I already enabled the page. Here is a more complete code snippet:
```
await Page.enable();
await Page.navigate({url: 'http://www.goodboydigital.com/pixijs/examples/12-2/'});
await Page.loadEventFired();
await Page.startScreencast({format: 'png', everyNthFrame: 1});
Page.screencastFrame( image =>{
          const {data, metadata} = image;
          console.log(metadata);
});
```

**Comment by auchenberg** [comment]
@t-io This seems like a problem with the Node client you are using. Please open the issue in the repo for the client, as this isn't a protocol issue.

**Comment by mafredri** [comment]
@t-io I was curious so I decided to implement this in my Go library (cdp) and I think I know why this is not working for you, there are two reasons:

1. After receiving a frame, you must acknowledge it (`Page.screencastFrameAck`) with the received session ID (e.g. do this in your `screencastFrame`-handler)
2. No events also means there will be no frames, try scrolling, highlighting, writing, etc. that will trigger new frames to be created

Implemented in Go, it would [look like this](https://github.com/mafredri/cdp/blob/80c11aba8ab177a877e3d618384e18635f9e63dc/example/screencast/main.go#L78-L82).

All this makes sense, from a protocol standpoint, on the other hand it's not very intuitive for a user, easy to miss.
